src/ore/src/operators_async_ext.rs

```diff
@@ -102,12 +106,16 @@ pub trait OperatorBuilderExt<G: Scope> {
     /// ecosystem the only way to yield control back to timely is by awaiting on
     /// `scheduler.yield_now()`. The operator will ensure that this call resolves when there is
     /// more work to do.
+    ///
+    /// The `reschedule_flag` argument to the `constructor` is used to communicate whether the
```

Comment:
I think we don't need to add this extra flag to `build_async`. As it is now, we expect the constructor `B` to return a future that returns `Never` (and so never resolves) but we also pass an `Rc<RefCell<bool>>` so that it can communicate when its done.

I think we should instead just remove the `Output = Never` requirement and just require that the constructor returns any `Fut: Future + 'static` and `.fuse()` it. Then, when we poll it we can decide whether to return true or false to `build_reschedule` depending on whether the poll was `Pending` or `Ready`. Using `.is_terminated()` of the fused future can also have an early return in case the logic future is done but for some reason there is more data on the inputs.